docs/middleware.md

```
@@ -0,0 +1,194 @@
+  if (context.operation === 'set' && value) {
+    return next(encrypt(value));
+  } else if (context.operation === 'get') {
+    const encrypted = await next(value);
+    return encrypted ? decrypt(encrypted) : undefined;
+  }
+  return next(value);
+});
+```
+
+In this example, when setting a value:
+1. The value is first compressed by the first middleware
```

markdownlint-cli2 (0.17.2)

Lists should be surrounded by blank lines
null

(MD032, blanks-around-lists)